docs/plugins.md

```diff
@@ -1 +1,124 @@
+  I recieved arguments: world
+```
+
+### Python
+
+As scripting languages Python and Ruby are ideal for workflow-type plugins and tooling and experimentation. The following plugins showcase the same plugin in both languages.
+
+The plugin `kn-health` uses `kubectl` to query the current Knative objects on a Knative installation for their health and summarizes the result. The goal is to give some idea of the healthiness of the Knative installation.
+
+A sketch of this plugin in Python follows. Complete [source code ](examples/plugins/python/kn-health.py) for the Python version.
+
+```python
+
+```
+
+### Ruby
+
+This is the same `kn-health` plugin written in Ruby. Complete [source code ](examples/plugins/ruby/kn-health.rb) for the Ruby version.
+
+```ruby
+
+```
+
+
+### Golang
+
+Since the Kn code base is itself in Golang, plugins written in this language are perhaps best for extensions. The resulting code can be structured like Kn itself which could lead to easier  possible integration into the Kn codebase in the future.
+
+The `kn-logs` tries to discover the pods for a running Knative service and aggregate the recent events and displays them to the user. The sketch of the Golang code follows with [complete source here](examples/plugins/golang/kn-logs).
+
+```golang
+
+```
+
+## Next steps
+
+The obvious next step for Kn plugins is to have users of Kn try and create their own plugins and provide feedback to the community under the `#kn-plugins` slack channel. However, there are more work to be done to make plugins usable for developers and users wanting to take Knative and Kn in production. 
+
+Specifically, we need a way to discover and manage plugins. Both at the level of the user’s local machine but also at a community level. Let’s explore each of these as potential future next steps and also discuss the notion of “server-side plugins”.
+
+### Discovery & Management
+
+You can discover the list of plugins visible in your local system by using the `kn plugin list` command. It will display all plugins visible to the current shell executing `kn` and you can ask it to change the location of where it looks for plugins and display more or less information as it searches for plugin. Please see the [kn plugin list](docs/cmd/kn_plugin_list.md) docs for details.
+
+However, what about discovering useful plugins from the community? Short of searching the internet or GitHub for Kn-related plugins it will be difficult for users of Knative and the Kn CLI to easily discover, share, and install remote plugins. This is not a new topic and the `kubectl` community has had to also address it. 
+
+The Kubernetes community has created the [`krew`](https://github.com/kubernetes-sigs/krew) package manager tool as one way to discover, package, and manage installation of `kubectl` plugins. We will be exploring whether extending `krew` is the best approach for Kn plugins or if we should create our own tool for discovery and managing plugins.
```

Comment:
Not sure about the scope of this document. Is it meant as a documentation for how plugins are currently working and how to implement them or more a roadmap document for what is planned ? Or a mixture ? Maybe we can split up the document in a pure reference documentation for describing what exists rights now and then have an additional kind of `plugin-roadmap.md` for additional ideas which might (or might not) be added.

Reply:
Yeah. I'll move the discovery part in a separate doc. I've been working on how to discover plugins but unable to spend the time on it to present to the team. Hopefully that changes soon ;) but in the meantime moving this to a roadmap doc makes sense. Thanks for the feedback.